packages/agreement-process/src/services/agreementService.ts

```diff
@@ -621,3 +638,82 @@ export async function cloneAgreementLogic({
     events: [newAgreement, ...docEvents],
   };
 }
+
+export async function rejectAreementLogic({
+  agreementId,
+  rejectionReason,
+  authData,
+  agreementQuery,
+  tenantQuery,
+  eserviceQuery,
+}: {
+  agreementId: string;
+  rejectionReason: string;
+  authData: AuthData;
+  agreementQuery: AgreementQuery;
+  tenantQuery: TenantQuery;
+  eserviceQuery: EserviceQuery;
+}): Promise<CreateEvent<AgreementEvent>> {
+  const agreementToBeRejected = await agreementQuery.getAgreementById(
+    agreementId
+  );
+  assertAgreementExist(agreementId, agreementToBeRejected);
+
+  assertRequesterIsProducer(
+    authData.organizationId,
+    agreementToBeRejected.data
+  );
+
+  assertExpectedState(agreementId, agreementToBeRejected.data.state, [
+    agreementState.pending,
+  ]);
+
+  const eservice = await eserviceQuery.getEServiceById(
+    agreementToBeRejected.data.eserviceId
+  );
+  assertEServiceExist(agreementToBeRejected.data.eserviceId, eservice);
+
+  const consumer = await tenantQuery.getTenantById(
+    agreementToBeRejected.data.consumerId
+  );
+  assertTenantExist(agreementToBeRejected.data.consumerId, consumer);
+
+  const descriptor = eservice.data.descriptors.find(
+    (d) => d.id === agreementToBeRejected.data.descriptorId
+  );
+  if (descriptor === undefined) {
+    throw descriptorNotFound(
+      eservice.data.id,
+      agreementToBeRejected.data.descriptorId
+    );
+  }
+
+  const stamp: AgreementStamp = {
+    who: authData.organizationId,
+    when: new Date(),
```

Comment:
In other parts of the code I see that stamps are created using `utcToZonedTime(new Date(), "Etc/UTC")`  instead of just `new Date()`, e.g., in `agreementSubmissionProcessor.ts`.
Is there a reason for this difference? Asking also because I am creating a stamp in another PR based on this one and need to understand the right way to do it :)

(cc @Viktor-K I remember you mentioned this in a conversation, if I'm not wrong)

Reply:
Correct 👍🏻  for date handling/creation we are using `date-fns-tz` util lib. I don't know in details if there was some strong reason to using `new Date()`, but it seems that the adoption of date-fns was overlooked.

Reply:
👍  I would double check also other places where stamps are created, I'll do it while working in #140 based on this

Reply:
Also, if I'm not wrong, looking at the Scala code (I'm also creating stamps in #140), shouldn't it be `authData.userId` instead of `authData.organizationId`?

Reply:
Confirmed, just double-checked Scala, in a further PR I am creating a utility for this: https://github.com/pagopa/interop-be-monorepo/pull/140/commits/c1c0a05ca425cc527290b9a7b0ce7713f859e5e3